### PR TITLE
Added ElementTransitionHasEnded condition

### PR DESCRIPTION
**Pull request by mnocon**
## Goal:
Add a new condition that allows us to detect in browser tests when a CSS transition has ended. An event listener is added that fires when a transition ends and adds a special class (`ibexa-selenium-transition-ended`) that we can later detect in tests.

https://user-images.githubusercontent.com/10993858/137711537-f74e5ca8-6d38-4956-93d2-8120a34914ad.mp4

## Example usage:
https://github.com/ezsystems/ezplatform-admin-ui/pull/1969/files

### https://github.com/ezsystems/BehatBundle/pull/220/commits/fe6b4bcd864d0871810db08ff8e1f3d301d13a45 - JavaScript changes

I've added the event listener and reorganised config manager to make it a bit more readable.
### https://github.com/ezsystems/BehatBundle/pull/220/commits/009bd1d29d1a44d130b54063366a8aba4ce57c09 - tests (PHP)

This PR is for master, because I want to make sure it solves the issue present there - if it works then I will backport it to 8.3 branch.

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/transition-listener.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const transitionEndedClass = 'ibexa-selenium-transition-ended'
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/transition-listener.js:1]:
```suggestion
const transitionEndedClass = 'ibexa-selenium-transition-ended';
```

File: src/bundle/Resources/public/js/scripts/transition-listener.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+document.addEventListener('transitionstart', function (e) { e.target.classList.remove(transitionEndedClass)})
+document.addEventListener('transitionend', function (e) { e.target.classList.add(transitionEndedClass)})
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/transition-listener.js:3]:
```suggestion
document.addEventListener('transitionend', (event) => event.target.classList.add(transitionEndedClass));
```

File: src/bundle/Resources/public/js/scripts/transition-listener.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const transitionEndedClass = 'ibexa-selenium-transition-ended'
+document.addEventListener('transitionstart', function (e) { e.target.classList.remove(transitionEndedClass)})
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/transition-listener.js:2]:
```suggestion
document.addEventListener('transitionstart', (event) => event.target.classList.remove(transitionEndedClass));
```

**Review by dew326 — APPROVED** [review]
just some nitpicks

**Review by tomaszszopinski — APPROVED** [review]
👍

**Pull request merged**